UofT-DSI | SQL - Assignment #2 Section 1, Section 2 and Section 3

AmmarMahzan opened the pull request:
<!-- PULL REQUEST TITLE: UofT-DSI | <Module Name> - Assignment <assignment number>-->

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

- Section 1: Logical Model ERD design and writeup
- Section 2: Advanced SQL queries (COALESCE, Windowed Functions, String Manipulations, UNION)
- Section 3: SQL operations (CROSS JOIN, INSERT, DELETE, UPDATE)

## What did you learn from the changes you have made?
Learned how to create logical models, manage changing data, and apply advanced SQL concepts like window functions and string manipulation.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
No other major approaches considered.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Faced small challenges handling CROSS JOIN explosion and UPDATE queries, especially referencing correct latest quantities, but managed to solve them.

## How were these changes tested?
Tested all queries in DB Browser for SQLite. Execution finished successfully.

Execution finished without errors.
Result: query executed successfully. Took 2ms
At line 227:
-- Update current_quantity from vendor_inventory
UPDATE product_units
SET current_quantity = (
  SELECT COALESCE(vi.quantity, 0)
  FROM vendor_inventory vi
  WHERE vi.product_id = product_units.product_id
  ORDER BY vi.market_date DESC
  LIMIT 1
)
WHERE EXISTS (
  SELECT 1 
  FROM vendor_inventory vi 
  WHERE vi.product_id = product_units.product_id
);

## A reference to a related issue in your repository (if applicable)

## Checklist
- [ x] I can confirm that my changes are working as intended

AmmarMahzan committed:
Add completed assignment1.sql and ERD diagram for Assignment 1

AmmarMahzan committed:
Completed Section 1 ERD and Assignment 2 SQL for Section 2 and 3

niyazmnazari reviewed (commented):
Thank you for submitting assignment 2. All queries are correct but you missed section 1.
Point: 20/70

AmmarMahzan commented:
Hi,

I have submitted section 1 together with all the other sections for
assignment 2. Attached is a copy of my submitted assignment 2 for section 1
which already committed in the github.

Thank you.

Ammar

On Tue, Apr 29, 2025 at 5:12 PM Niyaz Nazari ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Thank you for submitting assignment 2. All queries are correct but you
> missed section 1.
> Point: 20/70
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AmmarMahzan/https-github.com-UofT-DSI-sql/pull/2#pullrequestreview-2805097685>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BRHPCL2M3UYT2A6NWVZJL63237TMNAVCNFSM6AAAAAB37MQJACVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMBVGA4TONRYGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>

AmmarMahzan commented:
[AmmarMahzan_SQL_Assignment#2-Section1.pdf](https://github.com/user-attachments/files/19969120/AmmarMahzan_SQL_Assignment.2-Section1.pdf)

AmmarMahzan committed:
Update Assignment2.md

AmmarMahzan committed:
Add files via upload

Hi, I’ve updated Section 1 to align with the correct submission format. Previously, I submitted my written response as a PDF (AmmarMahzan_SQL_Assignment#2-Section1.pdf), but I now understand that the response should be written directly in Assignment2.md and supported by ERD diagrams. I’ve now moved the content accordingly and uploaded the required diagrams.

AmmarMahzan commented:
I’ve updated Section 1 to follow the correct format. The written content is now included directly in Assignment2.md, and all ERD diagrams have been uploaded as .png files in the same folder. This completes the Section 1 submission requirements.

niyazmnazari reviewed (approved):
Great job! 70/70

You have 4 'Sales' tables in first page and you fixed it in next pages.

AmmarMahzan commented:
Thank you. :)
